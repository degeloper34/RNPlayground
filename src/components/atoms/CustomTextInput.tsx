import {View, StyleSheet, TextInput, TextInputProps} from "react-native";
import Colors from "../../constants/Colors";

interface ITextInputProps extends TextInputProps {
  placeholder: string;
  backgroundColor?: string;
  textColor?: string;
}

export function CustomTextInput({
  backgroundColor = Colors.mainGrey,
  textColor = Colors.mainNight,
  ...attrs
}: ITextInputProps) {
  return (
    <View style={[styles.textInputView, {backgroundColor: backgroundColor}]}>
      <TextInput {...attrs} style={{color: textColor}} />
    </View>
  );
}

const styles = StyleSheet.create({
  textInputView: {
    justifyContent: "center",
    paddingHorizontal: 6,
    paddingVertical: 12,
    borderRadius: 8,
  },
});
