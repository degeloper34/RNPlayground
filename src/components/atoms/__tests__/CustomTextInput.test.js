import renderer from "react-test-renderer";
import React from "react";
import {CustomTextInput} from "../CustomTextInput";

describe("CustomTextInput", () => {
  it("username TextInput renders correctly", () => {
    const onChangeUsername = jest.fn();
    const component = renderer
      .create(
        <CustomTextInput
          placeholder="username"
          value={''}
          onChangeText={onChangeUsername}
          maxLength={30}
        />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("password TextInput renders correctly", () => {
    const onChangePassword = jest.fn();
    const component = renderer
      .create(
        <CustomTextInput
          placeholder="password"
          value={''}
          secureTextEntry
          onChangeText={onChangePassword}
          maxLength={30}
        />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
