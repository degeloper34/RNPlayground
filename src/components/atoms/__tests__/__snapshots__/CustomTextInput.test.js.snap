// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput password TextInput renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "borderRadius": 8,
      "flex": 1,
      "justifyContent": "center",
      "padding": 6,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={30}
    onChangeText={[MockFunction]}
    placeholder="password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`CustomTextInput username TextInput renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "borderRadius": 8,
      "flex": 1,
      "justifyContent": "center",
      "padding": 6,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={30}
    onChangeText={[MockFunction]}
    placeholder="username"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
